api/src/test/java/org/apache/iceberg/TestHelpers.java

```diff
@@ -137,6 +142,18 @@ public static void assertSerializedAndLoadedMetadata(Table expected, Table actua
     Assert.assertEquals("History must match", expected.history(), actual.history());
   }
 
+  public static void assertSerializedMetricsReporter(
+      MetricsReporter expected, MetricsReporter actual) {
+    Assert.assertNotNull("metrics reporter from serializableTable should not be null", actual);
+    Assert.assertTrue(
+        "metrics reporter from serializableTable should be instance of TestMetricsReporter",
+        actual instanceof TestMetricsReporter);
+    Assert.assertEquals(
+        "metrics reporter from serializableTable should equals the one from origin table",
+        expected,
+        actual);
```

Comment:
rather than testing actual equality on the objects, it should be enough to make sure that the properties are the same, hence the `TestMetricsReporter` wouldn't need to implement hasCode/equals

Reply:
done